gate-web/src/main/groovy/com/netflix/spinnaker/gate/controllers/V2PipelineTemplatesController.java

```diff
@@ -124,7 +124,28 @@ public Map get(@PathVariable String id) {
   public Map update(@PathVariable String id,
                     @RequestBody Map<String, Object> pipelineTemplate,
                     @RequestParam(value = "skipPlanDependents", defaultValue = "false") boolean skipPlanDependents) {
-    return null;
+    PipelineTemplate template;
+    try {
+      template = objectMapper.convertValue(pipelineTemplate, PipelineTemplate.class);
+    } catch (IllegalArgumentException e) {
+      throw new RuntimeException("Pipeline template is invalid", e);
```

Comment:
Does this end up being a 500 or a 400?

Reply:
This is currently 500 internal, which seems wrong for bad input. I'll see if I can rework this.

Reply:
You can use something like this: 
https://github.com/spinnaker/gate/blob/7e5a224690384ca9bb054df35ddd0d00b5b56f7a/gate-web/src/main/groovy/com/netflix/spinnaker/gate/controllers/PipelineController.groovy#L295-L307